### ISSUE TITLE
Trade Details

### ISSUE THREAD

**Issue by Bitoy** [OmniLayer/omnicore#256]
tx
bc4ca4a9b21675c8e5eea7d05884bf012f0dc98a97c81cd7870422ca64980fdf

In the Trade Details Window

Instead of 
Sell 1 OMNI for 4 SPT#12

Can it be 
Buy 4 SPT#12 for 1 OMNI

**Comment by dexX7** [comment]
Well, that's a bit tricky, because from the outer perspective "all trades are sell orders".

The issue is mostly one of words:

- Does Alice sell OMNI in exchange for BTC to Bob?
- Does Alice buy BTC in exchange for OMNI from Bob?
- Does Bob buy OMNI in exchange for BTC from Alice?
- Does Bob sell BTC in exchange for Omni from Alice?

**Comment by Bitoy** [comment]
ok "All trades are sell orders".

but it will be more intuitive if it is from perspective of the user.    
If Alice issued the buy order.  

In Alice Trade Details 
Buy 4 SPT#12 for 1 OMNI

In Bob's Trade Details it is
Sell 4 SPT#12 for 1 OMNI

**Comment by zathras-crypto** [comment]
Hey Bitoy,

Are you referring to the UI trade history?

Currently the data is populated from the transactions themselves - since a 'buy' is inverted into a sell by the MetaDEx interface before broadcast, the actual transaction carries no knowledge of that fact (there is no difference to any other sell offer) and so any interface querying 'history' data (UI or RPC) will only see it as a sell offer.

Basically, the 'buy' concept I put in the UI to simplify things for the end user, though I do worry about the confusion about buys being converted to sells sometimes!

**Comment by zathras-crypto** [comment]
> If Alice issued the buy order.
In Alice Trade Details
Buy 4 SPT#12 for 1 OMNI
In Bob's Trade Details it is
Sell 4 SPT#12 for 1 OMNI

It is feasible to do so, we would store a list of all the txids that came from our wallet that were buy offers.  I'd be against doing so though because it's something local only that can't be recreated from the state - a client upgrade for example that forces a ```--startclean``` would flip all existing buys to sells in the history likely freaking out the user hehe.

**Comment by Bitoy** [comment]
Hi Zathras

It is the UI under Exchange, Trade History.
 
A simple way to flip Sell to a Buy is to check if the first is token is an OMNI

ie. 
Sell 1 OMNI for 4 SPT#12

becomes
Buy 4 SPT#12 to 1 OMNI

(I dont think it will freak out the user =)

**Comment by dexX7** [comment]
MetaDEx was integrated some time ago.
